operator/tasks/allocator_aws/allocator_aws.go

```diff
@@ -12,17 +12,20 @@
 // See the License for the specific language governing permissions and
```

Comment:
Why not `pkg/ipam/allocator/aws/`?

Reply:
Good question! So it was a bit of dilemma for me... as I went on looking, and found `pkg/allocator`, `pkg/ipam/allocator` as well as `pkg/aws/eni`, and found it difficult to find the closes relationship, so kept it under `operator`. I am not attached to that decision, the only question I have is whether to keep `log.Fatal` or actually `return err`.

Do you think `pkg/ipam/allocator/aws` would be the best call? What about `pkg/aws/eni` or  `pkg/aws/eni/allocator`? Or actually, ` `pkg/aws/eip`? I am not quite sure why we refer to ENI and not EIP, as to me it's mostly about EIPs, or perhaps both to an equal extent, so perhaps `enieip`?

Reply:
I mean this is definitely related with "`ipam`" so IMO I think it should live under `pkg/ipam`. `pkg/allocator` as nothing to do with ipam since it's related with security identities. Now, is ENI related in anyhow with aws allocator?

Reply:
Living under `pkg/ipam/allocator/...` (where `...` is the cloud integration) LGTM.

Re: `eni` vs `eip` vs `aws`, I'd go for whatever is either cognitively easiest, easiest for developers to grep on / find references to, or whatever is more consistent across providers. Looks like `eip` is the vendor-specific naming, but `aws` is the vendor and AFAIK the IPAM on AWS is tied to the ENI devices in the APIs, so there may be an argument for consistency by calling it ENI, to make it slightly easier to find all AWS/ENI-related code.

Reply:
Ok, I'd go with `pkg/ipam/allocator/{aws,azure}`. I think `aws` is best, as we call Azure `azure` and not `NetworkInterface` (that is ENI equivalent in Azure.

Reply:
One question remains - do we keep `log.Fatal` or start returning errors and use `log.Fatal` in the caller?

Reply:
I would return an error. The caller might keep going and fall back to a different allocator for example

Reply:
For sure, I just wanted to clarify. We do have `log.Fatal` in some places in `pkg/`, but I that's probably unintentional and/or it's a technical debt.

Reply:
Yes, tech debt.